### ISSUE TITLE
[docs] Deploy SSR to AWS Amplify / Firebase 

### ISSUE THREAD

**Issue by W3Web** [quasarframework/quasar#6300]
I can’t deploy quasar ssr app to aws amplify or firebase.
Is that possible?

**Comment by TobyMosque** [comment]
create that files.:
**./.firebaserc**
```json
{
  "projects": {
    "default": "{project-name}"
  }
}
```
**./firebase.json**
```json
{
  "hosting": {
    "public": "dist/ssr",
    "ignore": [
      "firebase.json",
      "**/.*",
      "**/node_modules/**"
    ],
    "rewrites": [
      {
        "source": "**",
        "function": "ssrapp"
      }
    ]
  },
  "functions": {
    "source": "dist/ssr"
  }
}
```
**./src-ssr/__index.js**
```js
let server = require('./server/index.js')
exports.ssrapp = server.ssrapp
```
add the follow packages.:
```
yarn add firebase-admin firebase-functions
```
modify your **./src-ssr/index.js**
```js
/*...*/

const
  functions = require('firebase-functions'),
  express = require('express'),
  compression = require('compression')

const
  ssr = require('../ssr'),
  extension = require('./extension'),
  app = express()

const serve = (path, cache) => express.static(ssr.resolveWWW(path), {
  maxAge: cache ? 1000 * 60 * 60 * 24 * 30 : 0
})

// gzip
app.use(compression({ threshold: 0 }))

// serve this with no cache, if built with PWA:
if (ssr.settings.pwa) {
  app.use('/service-worker.js', serve('service-worker.js'))
}

// serve "www" folder
app.use('/', serve('.', true))

// we extend the custom common dev & prod parts here
extension.extendApp({ app })

// this should be last get(), rendering with SSR
app.get('*', (req, res) => {
  res.setHeader('Content-Type', 'text/html')
  ssr.renderToString({ req, res }, (err, html) => {
    if (err) {
      if (err.url) {
        res.redirect(err.url)
      } else if (err.code === 404) {
        res.status(404).send('404 | Page Not Found')
      } else {
        // Render Error Page or Redirect
        res.status(500).send('500 | Internal Server Error')
        if (ssr.settings.debug) {
          console.error(`500 on ${req.url}`)
          console.error(err)
          console.error(err.stack)
        }
      }
    } else {
      res.send(html)
    }
  })
})

// app.listen(port, () => {
//   console.log(`Server listening at port ${port}`)
// })
exports.ssrapp = functions.https.onRequest(app)
```

build the project, replace the `dist/ssr/index.js` by `src-ssr/__index.js` and deploy:
```
quasar build -m ssr
COPY src-ssr/__index.js dist/ssr/index.js
firebase deploy --project "{project-name}"
```

if u're using a CI service, get a token.:
```
firebase login:ci
```
and copy the follow command to your CI.:
```
firebase deploy --project "{project-name}" --token "{firebase-token}"
```

don't forget to replace `{project-name}` by the project name and `{firebase-token}` by he ci token

**Comment by riux** [comment]
`quasar build -m ssr`
I can't find the dist /server dir, I only have these directories

>dist
>>ssr
>>>www
>>>index.js
>>>package.json
>>>quasar.client-manifest.json
>>>quasar.server-manifest.json
>>>template.html

**Comment by pdanpdan** [comment]
dist/ssr :)

**Comment by riux** [comment]
@pdanpdan, What I can't understand is the part of: build the project, replace the `dist/ssr/index.js` by `src-ssr/__index.js` and deploy. 
if i replace `dist/ssr/index.js` with `src-ssr/__ index.js`

**Comment by danieltorscho** [comment]
Here is the updated solution I've came up with: https://stackoverflow.com/a/63843922/884143

**Comment by douglance** [comment]
@danieltorscho Does your solution for the `/src-ssr/index.js` include all the other express code from @TobyMosque's solution?  

I am getting 500s when deploying using the combination of both `index.js` files.

**Comment by danieltorscho** [comment]
> `QENV`

What does `QENV` mean ? quasar environement variable ?

**Comment by l1qu1d** [comment]
@TobyMosque or @danieltorscho with the new Quasar v2 they removed index.js in src-ssr folder and now use a middleware folder with a default render.js file. 

How can I get firebase and Quasar v2 ssr using expressJS to work? I've tried a lot of different things but the most logical, and according to the docs, would be to create a new middleware file using the command, `quasar new ssrmiddleware <name>`. 

The code I used in the created middleware file is:
```javascript
import { ssrMiddleware } from 'quasar/wrappers'
import * as functions from "firebase-functions";

// I export the app as a handler function from this file.
const app = require("../ssr/index.js");

export default ssrMiddleware(({ app }) => {
    exports.ssrapp = functions.https.onRequest(app.handler) 
})
```

I've also tried `exports.ssrapp = functions.https.onRequest(app)` in the render.js file as well with no success.

Build version: Quasar v2.0.0-rc.3
Quasar v2 SSR Middleware Documentation: [SSR Middleware](https://next.quasar.dev/quasar-cli/developing-ssr/ssr-middleware) 

I've been using firebase, Quasar v1 w/ ssr, and expresssJS successfully but with the changes in Quasar v2 I can't get it to work.

I appreciate the help!

**---EDIT & QUICK FIX---**

Found a solution and will write up a full explanation when I get free time.

In the meantime, for those who just want to quickly know what I had to do. I had to remove `module.exports = n` (line 140, 141 after adding exports.handler code) in the compiled ssr/index.js and added `exports.handler = h` (on line 139) in the IIFE where they had to "use strict".

** Final Edit **
Made a working ssr firebase repo with a full explanation.
https://github.com/l1qu1d/quasar-v2-ssr-firebase
